### PR TITLE
Add Automatic Cleaning Mode and Logging Features to Samsung AC Integration

### LINKED ISSUE #121: [NASA] Automatic Cleaning
Hello all,

I'd like to propose the addition of a toggle switch in this ESPHome integration for controlling the auto-cleaning function of air conditioning units. This feature allows the AC to continue running in fan-only mode for a few minutes after being turned off. Its primary purpose is to prevent the buildup of smells and mold inside the unit.

Currently, I can manage this functionality using the SmartThings HA integration along with a WiFi kit connected to my AC system. I believe this would be a valuable addition for others who wish to maintain their AC units more effectively.

I'm open to contribute to this feature by assisting with testing or exploring potential methods for enabling this functionality.

### PR DESCRIPTION
**Pull request by omerfaruk-aran**
This pull request introduces the automatic cleaning mode to the Samsung AC integration, enhancing the functionality and ensuring a more hygienic environment.

- Resolves #121

**Key Features:**

**- Automatic Cleaning Mode:**
 - Ensures hygienic drying of all moisture in the indoor unit after the cooling process is completed, providing an added layer of cleanliness and efficiency.

**- Logging Enhancements:**
 - **debug_log_messages:** This option allows for logging of all defined messages. When enabled (true), all standard log messages will be recorded, providing a clear overview of system operations.
 - **debug_log_undefined_messages:** This option logs any undefined messages, helping in debugging and identifying any unexpected or unhandled operations.
 - If both debug_log_messages and debug_log_undefined_messages are set to false, only the logs for the operations executed via SmartThings will be visible. This approach minimizes unnecessary log clutter while still providing essential logging for new features and functions.

**Changes Made Based on Previous Feedback:**
- Resolved all conflicts.
- Fixed indentation issues for better code readability.
- Renamed log parameters to start with debug_ for consistency with existing log options.
- Implemented the debug_log_messages functionality that was previously missing.
- Corrected the repository source address in the example YAML file.

This PR addresses and resolves the issues raised in the previous pull request and is now ready for review.

### CODE REVIEW

File: components/samsung_ac/protocol_non_nasa.cpp [hunk 1]
```diff
@@ -538,7 +538,7 @@ namespace esphome
 
         void process_non_nasa_packet(MessageTarget *target)
         {
-            if (debug_log_packets)
+            if (debug_log_messages)
```
Comment (lanwin) [inline components/samsung_ac/protocol_non_nasa.cpp:541]: This did not do what you would like to archive. This also print the undefined messages.

File: components/samsung_ac/protocol_nasa.cpp [hunk 2]
```diff
@@ -749,7 +870,7 @@ namespace esphome
 
             target->register_address(source);
 
-            if (debug_log_packets)
+            if (debug_log_messages)
```
Comment (lanwin) [inline components/samsung_ac/protocol_nasa.cpp:873]: This did not do what you would like to archive. This also print the undefined messages.

File: components/samsung_ac/protocol_nasa.cpp [hunk 3]
```diff
@@ -556,55 +569,88 @@ namespace esphome
             case MessageNumber::VAR_in_temp_room_f: //  unit = 'Celsius' from XML
             {
                 double temp = (double)message.value / (double)10;
-                ESP_LOGW(TAG, "s:%s d:%s VAR_in_temp_room_f %f", source.c_str(), dest.c_str(), temp);
+                if (debug_log_messages)
```
Comment (lanwin) [inline components/samsung_ac/protocol_nasa.cpp:572]: Adding the debug_log_message if to each case makes this already too large method even more larger. Can you please add and LOG_MESSAGE marco for that case?

**Review by lanwin — CHANGES_REQUESTED** [review]
Thank you for that pullrequest!

**Comment by omerfaruk-aran** [comment]
Requested changes have been implemented, please review the updates.

**Comment by lanwin** [comment]
Thank you!

**Comment by hnykda** [comment]
Wow, thanks @omerfaruk-aran . May I know how you actually figured this out? Could we use something similar for the #144 sound?

**Comment by lanwin** [comment]
@hnykda only when we found the corresponding messages...but for now we have no hint.

**Pull request merged**